File: appmonet-sdk/appmonet-sdk-dfp/src/main/java/com/monet/bidder/AdRequestFactory.java
package com.monet.bidder;

import androidx.annotation.NonNull;

import com.google.android.gms.ads.mediation.MediationAdRequest;
import com.monet.bidder.auction.AuctionRequest;

class AdRequestFactory {
  private AdRequestFactory() {
  }

  @NonNull
  static AdServerAdRequest fromAuctionRequest(boolean isPublisherAdView,
      AuctionRequest auctionRequest) {
    if (isPublisherAdView) {
      return DFPAdRequest.fromAuctionRequest(auctionRequest);
    }
    return DFPAdViewRequest.fromAuctionRequest(auctionRequest);
  }

  @NonNull
  static AdServerAdRequest createEmptyRequest(boolean isPublisherAdView) {
    if (isPublisherAdView) {
      return new DFPAdRequest();
    }
    return new DFPAdViewRequest();
  }

  @NonNull
  static AdServerAdRequest fromMediationRequest(boolean isPublisherAdView,
      @NonNull MediationAdRequest adRequest) {
    if (isPublisherAdView) {
      return new DFPAdRequest(adRequest);
    }

    return new DFPAdViewRequest(adRequest);
  }
}

File: appmonet-sdk/appmonet-sdk-dfp/src/main/java/com/monet/bidder/AdRequestFactory.kt
package com.monet.bidder

import com.google.android.gms.ads.mediation.MediationAdRequest
import com.monet.AdServerAdRequest
import com.monet.DFPAdRequest
import com.monet.DFPAdViewRequest
import com.monet.PublisherAdRequestWrapper
import com.monet.auction.AuctionRequest
import com.google.android.gms.ads.doubleclick.PublisherAdRequest.Builder
import com.monet.AdRequestWrapper
import com.monet.fromAuctionRequest

internal object AdRequestFactory {
  fun fromAuctionRequest(
    isPublisherAdView: Boolean,
    auctionRequest: AuctionRequest
  ): AdServerAdRequest {
    return if (isPublisherAdView) {
      DFPAdRequest.fromAuctionRequest(
          auctionRequest, MonetDfpCustomEventInterstitial::class.java,
          CustomEventBanner::class.java, CustomEventBanner::class.java,
          CustomEventInterstitial::class.java
      )
    } else DFPAdViewRequest.fromAuctionRequest(
        auctionRequest, MonetDfpCustomEventInterstitial::class.java,
        CustomEventBanner::class.java, CustomEventBanner::class.java,
        CustomEventInterstitial::class.java
    )
  }

  fun createEmptyRequest(isPublisherAdView: Boolean): AdServerAdRequest {
    return if (isPublisherAdView) {
      DFPAdRequest(PublisherAdRequestWrapper(Builder().build()))
    } else DFPAdViewRequest(
        AdRequestWrapper(com.google.android.gms.ads.AdRequest.Builder().build())
    )
  }

  fun fromMediationRequest(
    isPublisherAdView: Boolean,
    adRequest: MediationAdRequest
  ): AdServerAdRequest {
    return if (isPublisherAdView) {
      val dfpRequest = Builder()
          .setBirthday(adRequest.birthday)
          .setGender(adRequest.gender)
          .setLocation(adRequest.location)
          .build()
      DFPAdRequest(PublisherAdRequestWrapper(dfpRequest))
    } else {
      val req = com.google.android.gms.ads.AdRequest.Builder()
          .setBirthday(adRequest.birthday)
          .setGender(adRequest.gender)
          .setLocation(adRequest.location)
          .build()
      DFPAdViewRequest(AdRequestWrapper(req))
    }
  }
}